R/build-reference-index.R

```diff
@@ -42,30 +42,45 @@ data_reference_index_rows <- function(section, pkg) {
 
 
   if (has_name(section, "contents")) {
-    in_section <- select_topics(section$contents, pkg$topics)
-    topics <- pkg$topics[in_section, ]
-
-    contents <- tibble::tibble(
-      path = topics$file_out,
-      aliases = purrr::map2(
-        topics$funs,
-        topics$name,
-        ~ if (length(.x) > 0) .x else .y
-      ),
-      title = topics$title,
-      icon = find_icons(topics$alias, path(pkg$src_path, "icons"))
-    )
-
+    check_all_characters(section$contents, index, pkg)
+    contents <- purrr::imap(section$contents, content_info, pkg = pkg, section = index)
+    names <- unique(unlist(purrr::map(contents, "name")))
+    contents <- purrr::map(contents, function(x) x[names(x) != "name"])
+    contents <- do.call(rbind, contents)
     rows[[3]] <- list(
       topics = purrr::transpose(contents),
-      names = topics$name,
+      names = names,
       row_has_icons = !purrr::every(contents$icon, is.null)
     )
   }
 
   purrr::compact(rows)
 }
 
+check_all_characters <- function(contents, index, pkg) {
+  any_not_char <- any(purrr::map_lgl(contents, is_not_character))
+    if (any_not_char) {
+      abort(
+        paste(
+          sprintf(
+            "Content %s in section %s in %s must be a character.",
+            toString(which(any_not_char)),
+            index,
+            pkgdown_field(pkg, "reference")
+          ),
+          sprintf(
+            "%s You might need to add '' around e.g. - 'N' or - 'off'.",
+            crayon::bold("i")
+          ),
+          sep = "\n"
+        )
+      )
+    }
+}
+
+is_not_character <- function(x) {
+  !(typeof(x) == "character" && nzchar(x))
```

Comment:
Why check for empty strings here? It also the function name confusing because it's really checking that it's a string and it's not empty. It might be easier to just not make a function for this since you're only using it in one place and it's hard to give it a good name.